src/mrpro/operators/_Functional.py

```diff
@@ -0,0 +1,58 @@
+"""Base Class Functional."""
+
+# Copyright 2024 Physikalisch-Technische Bundesanstalt
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at:
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from abc import ABC
+from abc import abstractmethod
+from collections.abc import Sequence
+
+import torch
+
+from mrpro.operators._Operator import Operator
+
+
+class Functional(Operator[torch.Tensor, tuple[torch.Tensor]]):
+    """Functional Base Class."""
+
+    def __init__(
+        self, weight: torch.Tensor | float = 1.0, target: torch.Tensor | None = None, dim: Sequence[int] | None = None
+    ):
+        """Initialize a Functional.
+
+        Parameters
+        ----------
+            weight (float, optional) 
```

Comment:
we do not use optional or type definitions (e.g. float) because usually we forget to update them if we change the types (weight can also be a tensor not just a float).